cscs-checks/microbenchmarks/latency/src/latency.cpp

```diff
@@ -0,0 +1,230 @@
+// contributed by Sebastian Keller, CSCS, 2019-7
+
+#include <vector>
+#include <iostream>
+#include <algorithm>
+#include <chrono>
+#include <random>
+
+#include <thread>
+#include <pthread.h>
+
+#define PAGESIZE 4096
+#define CACHELINESIZE 64
+
+// only used for runtime estimation
+#define L1 32768
+#define L2 262144
+#define L3 25600*1024
+#define GHZ 3.3
```

Comment:
Why not guarding these with `ifndef`s so as to allow them to be defined during compilation per processor? Same for `PAGESIZE` and `CACHELINESIZE`.

Reply:
Since you're printing out the cycles based on the `GHZ`, and `PAGESIZE` and `CACHELINESIZE` are essential to your algorithm, I'd guard at least those with `ifndef`.

Reply:
ok